Assets/SEE/UI/PropertyDialog/UserPropertyDialog.cs

```diff
@@ -0,0 +1,162 @@
+        /// register on this event to receive a notification when this happens.
+        /// </summary>
+        public readonly UnityEvent OnCancel = new();
+
+
+
+        /// <summary>
+        /// The network configuration to be manipulated by this dialog.
+        /// </summary>
+        private Net.Network networkConfig;
+
+        /// <summary>
+        /// Creates and opens the dialog.
+        /// </summary>
+        public void Open()
+        {
+            networkConfig.Load();
+
+            dialog = new GameObject("User settings");
+            PropertyGroup group = dialog.AddComponent<PropertyGroup>();
+            group.Name = "User settings";
+            {
+                playerName = dialog.AddComponent<StringProperty>();
+                playerName.Name = "Username";
+                playerName.Value = networkConfig.Playername;
+                playerName.Description = "Username which will be used as avatar-tag and in other function.";
+                group.AddProperty(playerName);
+            }
+            // Dialog
+            propertyDialog = dialog.AddComponent<PropertyDialog>();
+            propertyDialog.Title = "User settings";
+            propertyDialog.Description = "Enter the user settings";
+            propertyDialog.AddGroup(group);
+
+            // Register listeners
+            propertyDialog.OnConfirm.AddListener(OKButtonPressed);
+            propertyDialog.OnCancel.AddListener(CancelButtonPressed);
+
+            SEEInput.KeyboardShortcutsEnabled = false;
+            // Go online
+            propertyDialog.DialogShouldBeShown = true;
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnCancel"/> and closes the dialog.
+        /// </summary>
+        private void CancelButtonPressed()
+        {
+            propertyDialog.Close();
+            OnCancel.Invoke();
+            SEEInput.KeyboardShortcutsEnabled = true;
+            Close();
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnConfirm"/> and save name and closes the dialog.
+        /// </summary>
+        private void OKButtonPressed()
+        {
+            bool errorOccurred = false;
+            {
+                string playername = playerName.Value.Trim();
+
+                if (ValidUsername(playername))
+                {
+                    networkConfig.Playername = playername;
+                }
+                else
+                {
+                    ShowNotification.Error
+                       ("Username Error",
+                        "Username needs to be minimum one character long.");
+                    errorOccurred = true;
+                }
+            }
+            if (!errorOccurred)
+            {
+                propertyDialog.Close();
+                OnConfirm.Invoke();
+                SEEInput.KeyboardShortcutsEnabled = true;
+                Close();
+                callBack?.Invoke();
+            }
```

Comment:
Instead of using the `errorOccurred` variable, consider unnesting the block containing the username check and just putting the contents of this if-block in the `else` block above.

Reply:
I used this way because it is the same as in NetworkPropertyDialog. Should i change both classes?

Reply:
Yes, please change both (at least in my opinion). I don't really see the point of the variable and I don't know why the code was this way in `NetworkPropertyDialog`.

Reply:
for the UserPropertyDialog i changed it. 

For NetworkPropertyDialog i let this unchanged because there a mutiple checks that can cause an error and i dont know how i can do it better without a variable that changes when some where a error occured.